packages/base/src/hooks/useSyncRef.ts

```diff
@@ -0,0 +1,19 @@
+import { MutableRefObject, Ref, RefCallback, RefObject, useRef } from 'react';
+
+export function useSyncRef<RefType = any>(ref: Ref<RefType>): [RefCallback<RefType>, RefObject<RefType>] {
```

Comment:
```suggestion
export function useSyncRef<RefType = any>(ref: Ref<RefType>): [RefCallback<RefType>, MutableRefObject <RefType>] {
```
isn't this rather a MutableRefObject?